### PR TITLE
update to ember-cli 3.26

### PR DESCRIPTION
**Pull request by Dhaulagiri**
### What this does do

* Update to ember-cli 3.26 and updates related packages based on blueprint updates
* Includes ember's new default prettier configuration and automatically updates most of the issues

### What this doesn't do
* Fix all the linting issues. I tried to fix the ones that seemed non-trivial but ones that seemed a bit harder I left as exceptions to be fixed later.
* Update to ember 3.26.  >3.25 caused issues with the global nav header where the dropdown examples no longer worked. Likely related to Ember integrating named blocks natively vs in the polyfill.

### CODE REVIEW

File: packages/pds-ember/addon/components/pds/app/body/index.hbs [hunk 1]
```diff
@@ -1,7 +1,6 @@
 <main
   class="pds-app__body"
   id="content"
-  role="main"
```
Comment (Dhaulagiri) [inline packages/pds-ember/addon/components/pds/app/body/index.hbs:4]: new template lint rules called this out as redundant on account of this already being a `<main>`

File: packages/pds-ember/addon/components/pds/breadcrumbs/index.hbs [hunk 2]
```diff
@@ -1,7 +1,6 @@
 <nav
   class="pds-breadcrumbs"
   aria-label={{t 'pds.components.breadcrumbs.aria.label'}}
-  role="navigation"
```
Comment (Dhaulagiri) [inline packages/pds-ember/addon/components/pds/breadcrumbs/index.hbs:4]: new template lint rules called this out as redundant on account of this already being a `<nav>`

File: packages/pds-ember/.template-lintrc.js [hunk 3]
```diff
@@ -1,5 +1,9 @@
 'use strict';
 
 module.exports = {
-  extends: 'octane'
+  extends: 'octane',
+
+  rules: {
+    'no-duplicate-landmark-elements': false,
```
Comment (meirish) [inline packages/pds-ember/.template-lintrc.js:7]: Why is this one turned off?

File: packages/pds-ember/addon/components/pds/app/index.hbs [hunk 4]
```diff
@@ -7,7 +7,7 @@
   Skip to Content
 </a>
 
-<a id="top" data-test-app-top-anchor></a>
+<a href="#" id="top" data-test-app-top-anchor></a>
```
Comment (meirish) [inline packages/pds-ember/addon/components/pds/app/index.hbs:10]: I think we want to put in an ignore for this one - the href-less is better than `href="#"` in this case imo.

File: packages/pds-ember/.template-lintrc.js (hunk 3 above)
Comment (Dhaulagiri) [reply packages/pds-ember/.template-lintrc.js:7]: The one addon code failure is [here](https://github.com/hashicorp/structure/blob/cebe144910ccef7040a19eb64ec49e7366398352/packages/pds-ember/addon/components/pds/global-header/index.hbs#L22) where there are multiple `nav` elements in the file depending on which conditional path is actually triggered (nested blocks strike again).  

There are a handful more in the dummy app so perhaps it'd be best to allow this in existing places but not add a global exception. What do you think?

Comment (johncowen) [reply packages/pds-ember/.template-lintrc.js:7]: You should be able to give multiple navs aria-labels:

https://github.com/hashicorp/consul/blob/cc5dba1e3b88cc670e75b1fafaf777536962e8c6/ui/packages/consul-ui/app/components/app/index.hbs#L57-L76

> If there is more than one navigation landmark role or <nav> element in a document, provide a label for each landmark.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Navigation_Role#labeling_landmarks

Not sure if the template linter follows those practices tho?

File: packages/pds-ember/addon/components/pds/app/index.hbs (hunk 4 above)
Comment (johncowen) [reply packages/pds-ember/addon/components/pds/app/index.hbs:10]: +1

File: packages/pds-ember/.eslintrc.js [hunk 5]
```diff
@@ -7,51 +7,52 @@ module.exports = {
...
+    'plugin:prettier/recommended',
   ],
   env: {
-    browser: true
+    browser: true,
   },
   rules: {
-    'ember/no-jquery': 'error'
+    'ember/no-jquery': 'error',
+    'ember/no-empty-glimmer-component-classes': 0,
```
Comment (johncowen) [inline packages/pds-ember/.eslintrc.js:24]: No biggie and I'm only guessing what this is for here. Wondering if we could just remove the empty glimmer component classes? Or I suppose if they are needed for a context in the template, maybe adding default values for the properties on the context might help?

File: packages/pds-ember/addon/components/pds/checkbox-field/index.js [hunk 6]
```diff
@@ -34,14 +34,15 @@ export default class PdsCheckboxField extends Component {
...
+      return this.args.id;
     }
 
     // otherwise, generate and memoize a guid
     if (!this._id) {
-      let guid = guidFor(this)
-      this._id = `chk-${guid}`
+      let guid = guidFor(this);
+      // eslint-disable-next-line ember/no-side-effects
+      this._id = `chk-${guid}`;
```
Comment (johncowen) [inline packages/pds-ember/addon/components/pds/checkbox-field/index.js:44]: There's a unique-id helper in the works somewhere that might help here (probably a future PR thing), we built our own until its available in core:

https://github.com/hashicorp/consul/blob/cc5dba1e3b88cc670e75b1fafaf777536962e8c6/ui/packages/consul-ui/app/helpers/unique-id.js

We inject our dom service, but you could just use `guidFor` straight in your helper if that feels better for you.

You could just `{{or @id (unique-id)}}` then.

File: packages/pds-ember/addon/components/pds/loading/elapsed/index.js [hunk 7]
```diff
@@ -14,20 +13,20 @@ import { tracked } from '@glimmer/tracking'
...
 function timeSince(startTime) {
-  let now = new Date().getTime()
-  let diff = now - startTime
+  let now = new Date().getTime();
+  let diff = now - startTime;
   // TODO: consider using date-fns (or JS:Temporal) to calculate this
-  let hours = Math.floor((diff % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60))
-  let minutes = Math.floor((diff % (1000 * 60 * 60)) / (1000 * 60))
-  let seconds = Math.floor((diff % (1000 * 60)) / 1000)
+  let hours = Math.floor((diff % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
```
Comment (johncowen) [inline packages/pds-ember/addon/components/pds/loading/elapsed/index.js:19]: @kaxcode oooh look, more mention of `temporal`

File: packages/pds-ember/tests/dummy/app/components/docs/global-header/index.hbs [hunk 8]
```diff
@@ -1,6 +1,7 @@
 <Pds::GlobalHeader>
   <:logo>
-  <img
+  <img {{!template-lint-disable no-potential-path-strings}}
+    alt="HashiCorp"
```
Comment (johncowen) [inline packages/pds-ember/tests/dummy/app/components/docs/global-header/index.hbs:4]: Wow TIL this one

**Review by johncowen — COMMENTED** [review]
Wow ❤️ `;`s 😄 

Thanks for doing this @Dhaulagiri ! I left a couple of comments, but mainly things that might help moving forwards. I didn't approve just as matt is on here and he's consuming this already

👏

File: packages/pds-ember/.eslintrc.js (hunk 5 above)
Comment (Dhaulagiri) [reply packages/pds-ember/.eslintrc.js:24]: _Probably_ but my memory is that there was a non-trivial number to address so it seemed worth punting on for now. I tried to get a more accurate number but for some reason I can no longer run `yarn lint:js` on this branch for reasons that I can't figure out right now 🤔 😭

File: packages/pds-ember/addon/components/pds/app/index.hbs (hunk 4 above)
Comment (Dhaulagiri) [reply packages/pds-ember/addon/components/pds/app/index.hbs]: This should be resolved now.

File: packages/pds-ember/.template-lintrc.js (hunk 3 above)
Comment (Dhaulagiri) [reply packages/pds-ember/.template-lintrc.js]: Pushed up some changes for this as well

Comment (johncowen) [reply packages/pds-ember/.template-lintrc.js]: Oh great, I guess the linter follows that practice then, excellent! I did notice one more tiny thing there:

```
aria-label="Primary navigation"
```

will read out "Primary navigation navigation" on a screen reader, so you might want to tweak that slightly?

> A brief description of the purpose of the navigation, omitting the term "navigation", as the screen reader will read both the role and the contents of the label.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Navigation_Role#associated_wai-aria_roles_states_and_properties

Comment (Dhaulagiri) [reply packages/pds-ember/.template-lintrc.js]: Good catch, thank you!

**Pull request merged**